### PR TITLE
ENH: added crs input to set_extent in examples

### PR DESCRIPTION
**Pull request by raybellwaves**
Fixes https://github.com/SciTools/cartopy/issues/925

### CODE REVIEW

**Comment by ajdawson** [comment]
Deleting the `__init__.py` is going to break things, as you can probably tell from the test output. You should remove that commit.

File: lib/cartopy/examples/reprojected_wmts.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ def plot_city_lights():
 
     # Plot WMTS data in a specific region, over a plain lat-lon map.
     ax = fig.add_subplot(1, 2, 1, projection=plain_crs)
-    ax.set_extent((-6, 3, 48, 58), crs=ccrs.PlateCarree())
+    ax.set_extent((-6, 3, 48, 58), crs=plain_crs)
```
Comment (ajdawson) [inline lib/cartopy/examples/reprojected_wmts.py]: This one doesn't need changing. For me this change reinforces the false perception that the projection and crs should be the same.

**Comment by raybellwaves** [comment]
@pelson did you receive my CLA and was in input correctly? I sent it in after my first PR. Cheers.

File: lib/cartopy/examples/eyja_volcano.py [hunk 2]
```diff
@@ -26,7 +26,7 @@ def main():
     ax = fig.add_subplot(1, 1, 1, projection=stamen_terrain.crs)
 
     # Limit the extent of the map to a small longitude/latitude range.
-    ax.set_extent([-22, -15, 63, 65])
+    ax.set_extent([-22, -15, 63, 65], crs=ccrs.Geodetic())
```
Comment (ajdawson) [inline lib/cartopy/examples/eyja_volcano.py]: Can you check this one @pelson

**Comment by raybellwaves** [comment]
Pinging @ajdawson

**Comment by corinnebosley** [comment]
@raybellwaves Can you rebase this please?  I don't think the failing tests are to do with your changes, so hopefully they should pass after a rebase.

**Comment by raybellwaves** [comment]
I started from scratch so closing this and opening another

**Pull request closed without merging**